src/thermo/Phase.cpp

```diff
@@ -573,7 +689,7 @@ void Phase::setConcentrations(const doublereal* const conc)
         norm += ck;
     }
     m_mmw = sum/norm;
-    setDensity(sum);
+    setConstDensity(sum);
```

Comment:
No. You could argue that there are liquid reacting phases that are nearly incompressible, where an update of the density is still appropriate.

Reply:
Sorry, I should have said for an *incompressible* phase, not constant density. For a simple liquid model, with a constant molar volume for each species, setting the mole fractions is of course fine, and would result in a change in density. However, setting the concentrations overspecifies things: either the density can be calculated from the species molar volumes, or the one determined from the concentrations and molecular weights alone.

Reply:
Of course, yes! It should definitely fail. Units are kmol/m3 for bulk phases ...

Reply:
@speth ... Implicitly, this appears to pull in #633. For any phase where `Phase::isCompressible` is `false` should trigger a recalculation of density, correct? Different from your suggestion in #633, this could now be addressed centrally?

Reply:
I'm not quite sure I see the relation. What is it that you think should trigger recalculation of density?

Reply:
A call to `Phase::setMoleFractions`. I guess I should have been more specific ...

Reply:
Yes, I think that would make sense, and agree that this PR isn't the place to do it. I'll take further discussion to #633 so it doesn't get lost here.